server/scripts/populate.py

```diff
@@ -1,138 +1,181 @@
+
 from configparser import ConfigParser
 from datetime import datetime
 from os import path
+from enum import Enum
+from marshmallow import Schema, fields, validate
 import pprint
 import uuid
-
 import psycopg2
-import yaml  # noqa
+import yaml
 
 filepath = "clubs.yml"
 pp = pprint.PrettyPrinter(indent=2)
 club_non_nullable = ["name", "description",
                      "size", "banner_image", "icon_image"]
 event_non_nullable = ["name", "description",
                       "banner_image", "start_time", "end_time"]
-tags_enum = ["Community", "Tech", "Creative", "Active", "Volunteering", "Gaming",
-             "Career", "Engineering", "Science", "Environment", "Arts", "Math", "Health"]
 tag_entities = {}
 
-# Assumptions: each club has its own events. Many-to-many not supported
-# yet. See example clubs.yml file for example input. db.ini file must be
-# in same directory
 
-def insert_tags(): 
-    conn = connect()
-    try:
-        with conn.cursor() as cursor:
-            for tag in tags_enum:
-                _id = str(uuid.uuid4())
-                tag_entities[tag] = _id
-                cursor.execute("INSERT INTO tag VALUES (%s,%s,%s,%s);",
-                               (_id, datetime.now(), datetime.now(), tag))
+class Tags(str, Enum):
+    community = "Community"
+    tech = "Tech"
+    creative = "Creative"
+    active = "Active"
+    volunteering = "Volunteering"
+    gaming = "Gaming"
+    career = "Career"
+    engineering = "Engineering"
+    science = "Science"
+    environment = "Environment"
+    arts = "Arts"
+    math = "Math"
+    health = "Health"
+
+
+class EventSchema(Schema):
+    name = fields.Str(required=True, validate=validate.Length(min=1))
+    description = fields.Str(required=True, validate=validate.Length(min=1))
+    url = fields.Url()
+    banner_image = fields.Url(required=True, validate=validate.Length(min=1))
+    start_time = fields.DateTime(required=True)
+    end_time = fields.DateTime(required=True)
+    facebook = fields.Url()
+    twitter = fields.Url()
+    instagram = fields.Url()
+    tags = fields.List(fields.Str(
+        validate=validate.OneOf([tag.value for tag in Tags])))
+
+
+class ClubSchema(Schema):
+    name = fields.Str(required=True, validate=validate.Length(min=1))
+    description = fields.Str(required=True, validate=validate.Length(min=1))
+    size = fields.Int(required=True, validate=validate.Range(min=1))
+    banner_image = fields.Url(required=True, validate=validate.Length(min=1))
+    icon_image = fields.Url(required=True, validate=validate.Length(min=1))
+    facebook = fields.Url()
+    twitter = fields.Url()
+    instagram = fields.Url()
+    website = fields.Url()
+    events = fields.List(fields.Nested(EventSchema))
+    tags = fields.List(fields.Str(
+        validate=validate.OneOf([tag.value for tag in Tags])))
+
+
+def insert_tags():
+    tag_args = []
+    for tag in Tags:
+        _id = str(uuid.uuid4())
+        tag_entities[tag.value] = _id
```

Comment:
🧙‍♀️
For even more cursed python magic, you can actually bake the ID right into the `Tags` enum, so that the `tag_entities` dictionary won't be necessary anymore.

(That's definitely optional though)